clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/TransformedCacheStoreClient.java

```diff
@@ -0,0 +1,55 @@
+package com.linkedin.davinci.client;
+
+import com.linkedin.venice.client.exceptions.VeniceClientException;
+import com.linkedin.venice.client.store.AvroGenericStoreClient;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import org.apache.avro.Schema;
+
+
+/**
+ * This class exposes a view of cache when customized transformation is enabled.
+ */
+public class TransformedCacheStoreClient<K, T> implements AvroGenericStoreClient<K, T> {
```

Comment:
Is K & T expected to be an Avro types/records?

Reply:
It would be great if T is not restricted to be Avro

Reply:
K is the same as the key inside the store, and `T` can be any type, which is the result of `CacheValueTransformer#transform`.

Reply:
There are pros and cons to having `T` be Avro or not... if it's Avro, then in DVC we could persist the output of the transformation and thus have quick bounce times, without losing local state. In many cases we could expect the transformed output to be smaller (e.g. in case of a projection of a subset of the fields) and so this also optimizes the size of the local state to the strict minimum required by the user. Of course the con is that it doesn't support arbitrary non-Avro objects.

And the pros and cons are the opposite if we support arbitrary transformation.

I would say it might be good to have both, actually. One way to look at it is that if the user opts for an Avro output, then we can give them extra optimizations.

Reply:
From LAN's code, it looks like the expected Quasar output for a given EntityKey would be:
Map<QuasarFeatureDefinition, Object>